File: algorithms-data-structures/ts/src/algorithms/mixed/reverse-in-parentheses/reverse.ts
export function reverseInParentheses(inputString: string): string {
  let stack: string[] = [];
  for (let ch of inputString) {
    if (ch === ")") {
      let temp: string[] = [];
      while (stack[stack.length - 1] !== "(") {
        let popped = stack.pop()!;
        temp.push(popped);
      }
      stack.pop();
      stack.push(...temp);
    } else {
      stack.push(ch);
    }
  }
  return stack.join("");
}

File: go/src/algorithms/mixed/reverse-in-parentheses/reverse.go
package reverseinparentheses

func reverseInParentheses(inputString string) string {
	stack := []rune{}
	for _, ch := range inputString {
		if ch == ')' {
			temp := []rune{}
			// Pop characters from stack till '(' is found
			for stack[len(stack)-1] != '(' {
				temp = append(temp, stack[len(stack)-1])
				stack = stack[:len(stack)-1]
			}
			// Pop the '('
			stack = stack[:len(stack)-1]
			// Push the reversed string back to stack
			stack = append(stack, temp...)
		} else {
			stack = append(stack, ch)
		}
	}
	return string(stack)
}
